### PR TITLE
Add `index` param for getting qss circuits

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Addresses https://github.com/Infleqtion/client-superstaq/pull/691#issuecomment-1682626197 and matches css implementation in #691

### CODE REVIEW

File: qiskit-superstaq/qiskit_superstaq/superstaq_job_test.py
Comment (SalahedeenIssa) [inline qiskit-superstaq/qiskit_superstaq/superstaq_job_test.py:212]: Is this necessary given that it's already tested with index=1 later on?

**Comment by SalahedeenIssa** [comment]
Other than that, lgtm! Maybe someone more familiar with the client could take a look if you wanna be sure.

File: qiskit-superstaq/qiskit_superstaq/superstaq_job_test.py
Comment (bharat-thotakura) [reply qiskit-superstaq/qiskit_superstaq/superstaq_job_test.py:212]: I don't think removing it will affect the coverage check, but I included line 212 to show `index` usage for a single `job_id` and line 222 to test `index` usage for a multi-`job_id`.

**Comment by bharat-thotakura** [comment]
Thanks for reviewing @SalahedeenIssa! Will wait for feedback on https://github.com/Infleqtion/client-superstaq/pull/691 to see if similar updates need to be made here as well before merging

File: qiskit-superstaq/qiskit_superstaq/superstaq_job.py
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_job.py:196]: nit
```suggestion
        if index is None:
            serialized_circuits = [self._job_info[job_id][circuit_type] for job_id in job_ids]
            return [qss.deserialize_circuits(serialized)[0] for serialized in serialized_circuits]

        gss.validation.validate_integer_param(index, min_val=0)
        serialized_circuit = self._job_info[job_ids[index]][circuit_type]
        return qss.deserialize_circuits(serialized_circuit)[0]
```
(this is super silly, but i always make silly mistakes when trying to follow the double negative in `if not .. else ..`)

**Review by richrines1 — COMMENTED** [review]
wdyt about adding an `index` argument to `result()` as well?

File: qiskit-superstaq/qiskit_superstaq/superstaq_job.py
Comment (bharat-thotakura) [reply qiskit-superstaq/qiskit_superstaq/superstaq_job.py:196]: No worries I think it is a fair point

**Comment by bharat-thotakura** [comment]
> wdyt about adding an `index` argument to `result()` as well?

That sounds good, and I've added a standard, simple approach in [094c6c6](https://github.com/Infleqtion/client-superstaq/pull/743/commits/094c6c69d2c219129017c36431b6d7ffa54f8a6d). Making more use of it, however, like you suggested here [094c6c6](https://github.com/Infleqtion/client-superstaq/pull/743/commits/094c6c69d2c219129017c36431b6d7ffa54f8a6d) will probably wait till another PR to limit this one's changes.

**Comment by richrines1** [comment]
lgtm!

**Review by richrines1 — APPROVED** [review]

**Pull request merged**
